File: backend/app/api/routes/notifications.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from typing import List
from uuid import UUID

from app.db.session import get_db
from app.api.deps import get_current_user
from app.models.user import User
from app.schemas.notification import NotificationRead
from app.services.notification_service import notification_service

router = APIRouter()


@router.get("/", response_model=List[NotificationRead])
def get_my_notifications(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return notification_service.get_for_user(db, current_user.id)


@router.get("/unread-count")
def get_unread_count(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    count = notification_service.get_unread_count(db, current_user.id)
    return {"count": count}


@router.patch("/{notification_id}/read", response_model=NotificationRead)
def mark_notification_read(
    notification_id: UUID,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    notif = notification_service.mark_read(db, notification_id, current_user.id)
    return notif


@router.post("/mark-all-read")
def mark_all_read(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    notification_service.mark_all_read(db, current_user.id)
    return {"message": "All notifications marked as read"}

File: frontend/src/api/notifications.ts
import { api } from './auth';
import type { Notification } from '../types';

export const notificationsService = {
    async getMyNotifications(): Promise<Notification[]> {
        const response = await api.get<Notification[]>('/notifications/');
        return response.data;
    },

    async getUnreadCount(): Promise<{ count: number }> {
        const response = await api.get<{ count: number }>('/notifications/unread-count');
        return response.data;
    },

    async markAsRead(id: string): Promise<Notification> {
        const response = await api.patch<Notification>(`/notifications/${id}/read`);
        return response.data;
    },

    async markAllAsRead(): Promise<{ message: string }> {
        const response = await api.post<{ message: string }>('/notifications/mark-all-read');
        return response.data;
    }
};

export default notificationsService;
